### ISSUE TITLE
Feature request: Local Echo

### ISSUE THREAD

**Issue by sk-jame** [fasteddy516/SimplySerial#38]
Hello, I think that very important feature is missing - local echo. I don't think it should be explained, it is well known interface. So just wanna mention that it would be great to have it here.

**Comment by salvathor79** [comment]
@sk-jame @lefebvresam 
Please see my pull request, I've just added this feature: #42

**Comment by lefebvresam** [comment]
When it's available in the installer?

**Comment by salvathor79** [comment]
@lefebvresam  I've uploaded a new version here
[v0.9.0](https://github.com/salvathor79/SimplySerial/releases/tag/v0.9.0)

**Comment by lefebvresam** [comment]
What also should be a nice to have is that you can add an arbitrary sequence when pressing enter. Because some command interpreters expect that you and a command sentence with 0xFF0xFF0xFF for instance.

**Comment by salvathor79** [comment]
@lefebvresam 
[v0.9.1](https://github.com/salvathor79/SimplySerial/releases/tag/v0.9.1)
In this version you can specify a custom string, the string is sent as-is. So no escape character or 0x... format value.

**Comment by lefebvresam** [comment]
Can you escape them, or use hex values? I want to send 0xFF0xFF0xFF for each string termination.

**Comment by lefebvresam** [comment]
But how do you enter  0xFF0xFF0xFF? As ÿÿÿ?

**Comment by salvathor79** [comment]
In the new version (v0.9.2) you can simply add the option **--tx:bytes="0xFF0xFF"** or  **--tx:bytes="FFFF"** or **--tx:bytes="FF FF"** or **--tx:bytes="0xFF 0xFF"** as you prefer.

note: **--tx** is the short form of **--tx_newline** option

**Comment by lefebvresam** [comment]
I use this which is working:

`ss -com:COM4 -echo:ON -tx:bytes="FFFFFF" -encoding:RAW -baud:120000`

But with the sniffer I see that every character is sent immedately after entering it.
When you have one char wrong you need te restart the whole line.
There is also a wait time between characters in my application before getting a timeout.
Should it be possible to send the whole line + termination only after pressing enter?
In that case it will be possible to edit the line before sending long strings.
Then it's more like a real terminal.

Thanks in advance.

**Comment by salvathor79** [comment]
@lefebvresam what you are asking is technical feasible, but this apparently simple modification is against the nature of Simply Serial that is a TERMINAL CONSOLE. In all terminal console programs like Simply Serial (i.e. Teraterm or Putty) each character is transmitted immediatelly, what you are asking is more a "serial tool" like Realterm or YAT.

If you want to correct keyboard mistakes, you can implement it in your application: simply upon receiving the char '\b', that is the backspace characters in C, you should remove the last char received before '\b' from the received string. Obviously this is only feasibile if you don't use any command that can contains the '\b' characther :-)

There are a lot more that you can do with escape sequence with a VT100 ANSI terminal like windows terminal and SimplySerial https://en.wikipedia.org/wiki/ANSI_escape_code

**Comment by fasteddy516** [comment]
The local echo feature has been added to the official SimplySerial codebase, and will be available in version 0.9.0 which should be released soon.  Note that I used @salvathor79's echo implementation pretty much as-is (thanks @salvathor79!), but did not merge in the full pull request as it contained other unrelated modifications that need further attention or to be implemented differently.
